src/status_im/switcher/switcher.cljs

```diff
@@ -1,44 +1,64 @@
 (ns status-im.switcher.switcher
-  (:require [quo.react-native :as rn]
-            [reagent.core :as reagent]
+  (:require [reagent.core :as reagent]
+            [quo2.reanimated :as reanimated]
             [status-im.switcher.styles :as styles]
-            [status-im.ui.components.animation :as anim]
             [status-im.switcher.animation :as animation]
             [status-im.ui.components.icons.icons :as icons]
             [status-im.react-native.resources :as resources]
             [status-im.switcher.switcher-container :as switcher-container]))
 
-(defn toggle-switcher-screen [switcher-opened? view-id anim-values]
-  (swap! switcher-opened? not)
-  (animation/animate @switcher-opened? view-id anim-values))
+(defn switcher-button
+  [view-id close-button-opacity switcher-button-opacity
+   button-touchable-scale toggle-switcher-screen-fn]
+  [:>
+   (fn []
+     (let [touchable-original-style       (styles/switcher-button-touchable view-id)
+           close-button-original-style    (styles/switcher-close-button)
+           switcher-button-original-style (styles/switcher-button)
+           touchable-animated-style       (reanimated/apply-animations-to-style
+                                           {:transform [{:scale button-touchable-scale}]}
+                                           touchable-original-style)
+           close-button-animated-style    (reanimated/apply-animations-to-style
+                                           {:opacity close-button-opacity}
+                                           close-button-original-style)
+           switcher-button-animated-style (reanimated/apply-animations-to-style
+                                           {:opacity switcher-button-opacity}
+                                           switcher-button-original-style)]
+       (reagent/as-element
```

Comment:
I really liked reagent's shorthand `[:f>` for creating [functional components](https://reactjs.org/docs/components-and-props.html#function-and-class-components), but for some reason it is not working, (only in android PR build) and giving runtime error `TypeError: Cannot read property 'indexOf' of null`.

Reply:
could you provide an example ? 

also you could try to set
```
:static-fns true
:fn-invoke-direct true
```

Reply:
> could you provide an example ?

```
(defn child-comp []
  [:f> (fn []
         [react/view])])
```

If I use above child comp inside any parent component, then the PR build app crashes at runtime with `TypeError: Cannot read property 'indexOf' of null`, but If I use

```
(defn child-comp []
  [:> (fn []
        (reagent/as-element
         [react/view]))])
```

Then it just works, not sure why first one is failing

> also you could try to set
> ```
> :static-fns true
> :fn-invoke-direct true
> ```

it is already there, https://github.com/status-im/status-react/blob/9f9b896eb7fcfac60d55b33a3744480b13d2ba8d/shadow-cljs.edn#L64

Reply:
i mean in :dev build. to see if its a reason

Reply:
**Fixed**: _Adding solution for reference_

CI build was using a different version of the reagent (0.10.0), than the dev build(1.0.0). And `:f>` is only available in `1.0.0+` versions.
`make nix-update-clojure` fixed the issue.

Thank you @flexsurfer and @Jakubgs